File: client/src/Types/ClinicalRecord.ts
import type { Patient } from "./Patient.ts";
import type { Doctor } from "./Doctor.ts";

export type ClinicalRecord = {
  id: number;
  patient: Patient;
  recordedByDoctor: Doctor;
  recordedDate : string;
  chestPainType: "Typical Angina" | "Atypical Angina" | "Non-Anginal Pain" | "Asymptomatic";
  restingBloodPressure: number;
  cholesterolTotal: number;
  fastingBloodSugar: "Greater than or equal to 120 mg/dl" | "Less than 120 mg/dl";
  restEcg: "Normal" | "Having ST-T wave abnormality" | "Showing probable or definite left ventricular hypertrophy by Estes' criteria";
  maximumHeartRate: number;
  exerciseInducedAngina: boolean;
  oldPeak : number;
  slope: "Upsloping" | "Flat" | "Downsloping";
  majorVesselsColored: 0 | 1 | 2 | 3;
  thalassemia: "Error" | "Normal" | "Fixed Defect" | "Reversible Defect";
  probability: number;
  label: boolean;
}

File: server/HealthcareApi/Models/ClinicalRecord.cs
using HealthcareApi.Enums;

namespace HealthcareApi.Models;

public class ClinicalRecord
{
  public int Id { get; set; }
  public Patient? Patient { get; set; }
  public int PatientId { get; set; }
  public Doctor? RecordedByDoctor { get; set; }
  public int RecordedByDoctorId { get; set; }
  public DateTime RecordedDate { get; set; } = DateTime.UtcNow;
  public ChestPainType ChestPainType { get; set; }
  public int RestingBloodPressure { get; set; }
  public int CholesterolTotal { get; set; }   //serum cholesterol in mg/dl
  public bool FastingBloodSugar { get; set; }  //fasting blood sugar > 120 mg/d
  public RestECG RestEcg { get; set; }
  public int MaximumHeartRate { get; set; }
  public bool ExerciseInducedAngina { get; set; }
  public double OldPeak { get; set; } //ST depression induced by exercise relative to rest
  public Slope Slope { get; set; } // the slope of the peak exercise ST segment
  public int MajorVesselsColored { get; set; } // 0-3
  public Thalassemia Thalassemia  { get; set; }
  public bool Label { get; set; }
  public double Probability { get; set; }
}
